backup.sh

```diff
@@ -1,145 +1,228 @@
+    echo "Replacing ${INCREMENT} backup ${DEST} file this last backup..."
+    mv "${DESTNEW}" "${DEST}"
   else
-    echo "Replacing daily backup ${DFILE} file this last backup..."
-    ln -vf "${FILE}" "${DFILE}"
-    echo "Replacing weekly backup ${WFILE} file this last backup..."
-    ln -vf "${FILE}" "${WFILE}"
-    echo "Replacing monthly backup ${MFILE} file this last backup..."
-    ln -vf "${FILE}" "${MFILE}"
+    echo "Replacing ${INCREMENT} backup ${DEST} file this last backup..."
+    ln -vf "${SRC}" "${DEST}"
   fi
   # Update latest symlinks
-  echo "Point last backup file to this last backup..."
-  ln -svf "${LAST_FILENAME}" "${BACKUP_DIR}/last/${DB}-latest${BACKUP_SUFFIX}"
-  echo "Point latest daily backup to this last backup..."
-  ln -svf "${DAILY_FILENAME}" "${BACKUP_DIR}/daily/${DB}-latest${BACKUP_SUFFIX}"
-  echo "Point latest weekly backup to this last backup..."
-  ln -svf "${WEEKLY_FILENAME}" "${BACKUP_DIR}/weekly/${DB}-latest${BACKUP_SUFFIX}"
-  echo "Point latest monthly backup to this last backup..."
-  ln -svf "${MONTHY_FILENAME}" "${BACKUP_DIR}/monthly/${DB}-latest${BACKUP_SUFFIX}"
-  #Clean old files
-  echo "Cleaning older files for ${DB} database from ${POSTGRES_HOST}..."
-  find "${BACKUP_DIR}/last" -maxdepth 1 -mmin "+${KEEP_MINS}" -name "${DB}-*${BACKUP_SUFFIX}" -exec rm -rvf '{}' ';'
-  find "${BACKUP_DIR}/daily" -maxdepth 1 -mtime "+${KEEP_DAYS}" -name "${DB}-*${BACKUP_SUFFIX}" -exec rm -rvf '{}' ';'
-  find "${BACKUP_DIR}/weekly" -maxdepth 1 -mtime "+${KEEP_WEEKS}" -name "${DB}-*${BACKUP_SUFFIX}" -exec rm -rvf '{}' ';'
-  find "${BACKUP_DIR}/monthly" -maxdepth 1 -mtime "+${KEEP_MONTHS}" -name "${DB}-*${BACKUP_SUFFIX}" -exec rm -rvf '{}' ';'
-done
-
-echo "SQL backup created successfully"
-
-# Post-backup hook
-if [ -d "${HOOKS_DIR}" ]; then
-  run-parts -a "post-backup" --reverse --exit-on-error "${HOOKS_DIR}"
-fi
+  echo "Replacing lastest ${INCREMENT} backup to this last backup..."
+  ln -svf "${DEST}" "${BACKUP_DIR}/${INCREMENT}/${DB}-latest"
+
+}
+
+#Clean up old backups
+cleanup_backups () {
+  
+  for folder in "${FREQUENCY[@]}"
+  do
+    if [ $folder == 'last' ]
+    then
+      KEEP=$KEEP_MINS
+    elif [ $folder == 'daily' ]
+    then
+      KEEP=$KEEP_DAYS
+    elif [ $folder == "weekly" ]
+    then
+      KEEP=$KEEP_WEEKS
+    elif [ $folder == 'monthly' ]
+    then
+      KEEP=$KEEP_MONTHS
+    fi
+
+    for DB in ${POSTGRES_DBS}
+    do
+
+      #Clean old files
+      local all=( `find "${BACKUP_DIR}/${folder}" -maxdepth 1 -name "${DB}-*"` )
+      local files=()
+
+      if [ $KEEP -gt 0 ]
+      then
+      
+        echo "Cleaning older files in ${folder} for ${DB} database from ${POSTGRES_HOST}..."
+        local files=( `find "${BACKUP_DIR}/${folder}" -maxdepth 1 -mtime "+$((${KEEP}-1))" -name "${DB}-*"` )
+        local files=( `printf "%s\n" "${files[@]}" | sort -t/ -k3` )
+
+      fi
+
+  	  for file in "${files[@]}"
+      do
+
+        echo "Deleting $file"
+  		  rm $file
```

Comment:
This might fail in some use cases.
In some use cases the `$file` will be a non empty folder.

Reply:
fixed this